lib/internal/Magento/Framework/View/Result/Layout.php

```diff
@@ -7,6 +7,7 @@
 namespace Magento\Framework\View\Result;
 
 use Magento\Framework;
+use Magento\Framework\App\Response\HttpInterface as HttpResponseInterface;
```

Comment:
`ResultInterface` is not HTTP agnostic at the moment at least because it has `setHttpResponseCode` method. Looks like we need to investigate more how exactly this should be refactored/fixed (created internal ticket MAGETWO-52704)